### PR TITLE
Added support for Agent Metrics

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
Fixes https://github.com/G-Research/consuldotnet/issues/262
@marcin-krystianc  @naskio

### CODE REVIEW

File: Consul.Test/AgentTest.cs [hunk 1]
```diff
@@ -993,5 +993,12 @@ public async Task Agent_FilterChecks()
             Assert.Single(actual);
             Assert.Equal(checkName, actual.Values.First().Name);
         }
+
+        [Fact]
+        public async Task Agent_Metrics()
+        {
+            var agentMetrics = await _client.Agent.GetAgentMetrics();
+            Assert.NotNull(agentMetrics.Response.Timestamp);
```
Comment (marcin-krystianc) [inline Consul.Test/AgentTest.cs:1001]: I don't see a reason why not to assert, that all other members are not null:
```suggestion
            Assert.NotNull(agentMetrics.Response.Timestamp);
            Assert.NotNull(agentMetrics.Response.Counters);
            Assert.NotNull(agentMetrics.Response.Gauges);
            Assert.NotNull(agentMetrics.Response.Points);
            Assert.NotNull(agentMetrics.Response.Samples);
```

**Review by marcin-krystianc — COMMENTED** [review]
I think it looks good, just left one comment regarding the test.

**Review by marcin-krystianc — APPROVED** [review]
Thank you 👍

**Comment by sammychinedu2ky** [comment]
My pleasure

**Pull request merged**